### PR TITLE
Compile with mutter-8 (GNOME 40 beta)

### PR DESCRIPTION
**Pull request by fossfreedom**
## Description
This adds mutter-8 (GNOME-40) compatibility.

~Note - this is very early work not ready for merging until at least GNOME 40 (beta) is reached.~
PR is for anyone who is interested in helping to develop and test ~pre-~beta upstream work

Tested mutter tagged 40.alpha1.1 & 40.beta  Changes:

1. Add mutter-8 compilation vapi
2. Fixed vapi warnings for mutter 6/7 & 8 - better custom vapi to ensure extraction of API and mapping to mutter c header files
2. added HAVE_MUTTER_8 conditional
3. conditional needed on shim.vala - keyboard shortcuts/custom and remapping needs careful testing

### Submitter Checklist

- [ ] Squashed commits with `git rebase -i` (if needed)
- [ ] Built budgie-desktop and verified that the patch worked (if needed)

### CODE REVIEW

**Comment by fossfreedom** [comment]
@JoshStrobl FYI https://discourse.ubuntu.com/t/staying-on-gtk3-and-gnome-3-38-this-cycle/20466

TLDR - looks like Ubuntu will not be moving beyond the current GNOME stack until at least Sept this year.  Probably this will include mutter.  Certainly Debian will not see GNOME 40 stuff until after their freeze much later this year.

As such I will continue to look at what needs doing for mutter/budgie when mutter-8 hits beta but I will not be able to fully run through stuff because we won't have the GNOME stack to play with.

I don't know what your Solus plans are for April onwards - but you may want to consider those thoughts on the above thread.

**Comment by JoshStrobl** [comment]
Hey @fossfreedom, thanks for the patch. I will be continuing to monitor the ABI changes, when and should they occur, for Mutter 8, as it is my plan to *eventually* move Solus to GNOME 40 when there is an appropriate time to do so.

Like Ubuntu, I share a concern about the potential breakages in GNOME 40 related to the Shell. That being said, I typically exercise that level of caution with *every* new GNOME Stack upgrade, as .0 releases from them are typically bug-ridden. It wouldn't be too much of a trouble for me to build against any beta releases of the various parts of the stack in a separate Solus binary package repo and get Budgie functional there before shipping the patches here, so there shouldn't be any concern on Ubuntu Budgie's part as to the support of GNOME 40 stack in Budgie Desktop when Ubuntu decides to move to it :)

**Comment by gmbr3** [comment]
I have started testing this for openSUSE as we already have a GNONE testing repo set up for GNOME 40

**Comment by rodrigofarias77** [comment]
I maintain a Copr to build budgie-desktop for Fedora folks.

rawhide's mutter 40.0~alpha.1.1-1.fc34 is apparently happy with this.

**Comment by rodrigofarias77** [comment]
Looks good with Fedora's mutter 40.0~alpha.1.1-4.20210202gita9d9aee6c.fc34. Getting the Copr updated now.

**Comment by JoshStrobl** [comment]
The patch looks goo to me, thanks!
